src/DataStructures/Tensor/EagerMath/OrthonormalOneform.hpp

```diff
@@ -0,0 +1,74 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+
+/// \cond
+namespace gsl {
+template <typename>
+struct not_null;
+}  // namespace gsl
+/// \endcond
+
+// @{
+/*!
+ * \ingroup TensorGroup
+ *
+ * \brief Compute a spatial one-form orthonormal to the given unit form
+ *
+ * Given a unit spatial one-form \f$s_i\f$, compute a new form \f$t_i\f$
+ * which is orthonormal to \f$s_i\f$, in the sense that
+ * \f$\gamma^{ij}s_i t_j = 0\f$, for the given inverse spatial metric
+ * \f$\gamma^{ij}\f$.
+ *
+ * \details The new form is obtained via Gram-Schmidt process, starting
+ * from a form whose components are \f$t_i = \delta_i^I\f$, where \f$I\f$ is
+ * the index of the smallest absolute value of the components of \f$s_i\f$.
+ */
+template <typename DataType, size_t VolumeDim, typename Frame>
+void orthonormal_oneform(
+    gsl::not_null<tnsr::i<DataType, VolumeDim, Frame>*> orthonormal_form,
+    const tnsr::i<DataType, VolumeDim, Frame>& unit_form,
+    const tnsr::II<DataType, VolumeDim, Frame>& inv_spatial_metric) noexcept;
+
+template <typename DataType, size_t VolumeDim, typename Frame>
+tnsr::i<DataType, VolumeDim, Frame> orthonormal_oneform(
+    const tnsr::i<DataType, VolumeDim, Frame>& unit_form,
+    const tnsr::II<DataType, VolumeDim, Frame>& inv_spatial_metric) noexcept;
+// @}
+
+// @{
+/*!
+ * \ingroup TensorGroup
+ *
+ * \brief Compute a spatial one-form orthonormal to the given unit forms
```

Comment:
I suggest saying "...orthonormal to two given unit forms"  to make the distinction from the other function clearer. Up to you if you want to make this change :)